packages/mobile/src/exchange/WithdrawCeloScreen.tsx

```diff
@@ -43,11 +55,25 @@ function WithdrawCeloScreen({ navigation }: Props) {
     celoToTransferNumber.isGreaterThan(0) &&
     celoToTransferNumber.isLessThanOrEqualTo(goldBalanceNumber)
 
-  const onConfirm = () => {
+  const onConfirm = async () => {
+    setLoading(true)
     const celoAmount = new BigNumber(celoToTransfer)
+    const cUsdAmount = await celoToUsd(celoAmount)
+
+    const now = Date.now()
+    const isLimitReached = isPaymentLimitReached(now, recentPayments, cUsdAmount.toNumber())
+    if (isLimitReached) {
+      dispatch(
+        showLimitReachedError(now, recentPayments, localCurrencyExchangeRate, localCurrencySymbol)
+      )
+      setLoading(false)
+      return
+    }
+
```

Comment:
Can we organize this into a separate function?

Reply:
Yup, probably a good idea to reuse this logic between this file and the one in the Send flow too.